web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -377,6 +404,66 @@ private void authUpdate(HandshakeResponse handshakeResponse) {
         }, 2500);
     }
 
+    private void subscribeToTerminationNotification() {
+        sessionServiceClient.terminationNotification(new TerminationNotificationRequest(), metadata(),
+                (fail, success) -> {
+                    if (fail != null) {
+                        if (checkStatus((ResponseStreamWrapper.Status) fail)) {
+                            // restart the termination notification
+                            subscribeToTerminationNotification();
+                            return;
+                        }
+                    }
+
+                    // welp; the server is gone -- let everyone know
+                    info.notifyConnectionError(new ResponseStreamWrapper.Status() {
+                        @Override
+                        public double getCode() {
+                            return 14; // UNAVAILABLE
+                        }
+
+                        @Override
+                        public String getDetails() {
+                            final TerminationNotificationResponse response = (TerminationNotificationResponse) success;
+                            if (!response.getAbnormalTermination()) {
+                                return "Server exited normally.";
+                            }
+
+                            final StringBuilder retval;
+                            if (!response.getReason().isEmpty()) {
+                                retval = new StringBuilder(response.getReason());
```

Comment:
fun fact, the gwt impl of stringbuilder/buffer is just a string that gets stuff +='d to it, its more efficient in all modern browsers than joining an array after you know the full size